### ISSUE TITLE
Microsoft Edge failed update

### ISSUE THREAD

**Issue by igorskyflyer** [microsoft/winget-pkgs#15507]
I checked for app updates using the CLI, it showed a few, I ran the **upgrade** command for each manually and all of them have been updated successfully, all but Microsoft Edge (from 90.x.xxx.xx -> 91.0.864.37). The command **winget list** shows that there is a new version available with the Source column value being set to "winget" but when I try to upgrade it shows there are no sources available, the same happens with the **upgrade --all** command and switches.

Environment info:
Edition	Windows 10 Pro
Version	21H1
Installed on	‎22/‎05/‎21
OS build	19043.985
Experience	Windows Feature Experience Pack 120.2212.2020.0

**Comment by jedieaston** [comment]
I don't think (someone correct me if I'm wrong) that the Edge msi can be used for upgrades. It always throws a 1603 for me if Edge has already been installed (probably something to do with the MSI installing an appx package, but honestly that's just a guess.

**Comment by denelon** [comment]
@jedieaston, I think we have some work to do with MSI in the upgrade scenario. I believe (just a guess) if it were an MSIX it would just gracefully upgrade. With MSI there is an UpgradeCode we need to pull and look at better behaviors.

**Comment by denelon** [comment]
I believe Edge will upgrade itself. If "requireExplicitUpgrade" is added to the manifest, it will be ignored by `winget upgrade --all`.

**Comment by igorskyflyer** [comment]
> I believe Edge will upgrade itself. If "requireExplicitUpgrade" is added to the manifest, it will be ignored by `winget upgrade --all`.

True, Edge (as other Chromium-based browsers) auto-upgrades in the background but if it is listed as upgradable then it should be, if not - just ignored for upgrading completely. 😄

**Comment by Witchilich** [comment]
```
❯ winget upgrade Microsoft.Edge
A newer version was found, but the install technology is different from the current version installed. Please uninstall the package and install the newer version.
```
Unfortunately I can't uninstall Microsoft Edge in Windows 11.
```
❯ winget uninstall Microsoft.Edge
Found Microsoft Edge [Microsoft.Edge]
Starting package uninstall...
Uninstall failed with exit code: 93
```
I was trying to uninstall it, so that I could just install Microsoft Edge from the msstore. But I can't.

**Comment by igorskyflyer** [comment]
> 

Well, I mean, even though it's a workaround but you can always uninstall it via Apps in the Settings. Let me know if that works. 🧘🏻‍♂️

**Comment by xIUPITERx** [comment]
> 
> Unfortunately I can't uninstall Microsoft Edge in Windows 11.
> 
> I was trying to uninstall it, so that I could just install Microsoft Edge from the msstore. But I can't.

The problem is, that winget uses the Microsoft Edge Enterprise version, which is not compatible with the normal Chromium Edge version for private use.

>
````
PS C:\Users\USER> winget show Microsoft.Edge
Found Microsoft Edge [Microsoft.Edge]
Version: 110.0.1587.69
Publisher: Microsoft Corporation
Publisher Url: https://www.microsoft.com/en-US/
Publisher Support Url: https://support.microsoft.com/en-us/microsoft-edge
Author: Microsoft Corporation
Moniker: microsoft-edge
Description: World-class performance with more privacy, more productivity, and more value while you browse.
Homepage: https://www.microsoft.com/en-us/edge
License: MIT License
License Url: https://www.microsoft.com/en-us/servicesagreement/
Privacy Url: https://privacy.microsoft.com/en-US/privacystatement
Copyright: Copyright (C) Microsoft Corporation
Copyright Url: https://www.microsoft.com/en-us/servicesagreement/
Tags:
  blink
  browser
  chromium
  edge
  web
  web-browser
Installer:
  Installer Type: wix
  Installer Url: https://msedge.sf.dl.delivery.mp.microsoft.com/filestreamingservice/files/ec39329e-4889-4526-bee3-76706652f0b8/MicrosoftEdgeEnterpriseX64.msi
  Installer SHA256: dcb6413b96338e9dee5ae28385a11bd968076550c42f05e2c5264e4d7f86f695
````
> ``MicrosoftEdgeEnterpriseX64.msi``

**Comment by blap** [comment]
I still got this problem...
Any help?

**Comment by igorskyflyer** [comment]
Unfortunately not, I switched to only manual updates of Edge - via Edge itself - in the Settings of it. 🙃

**Comment by alanlivio** [comment]
Hi, I having the same error today:

```
>     winget upgrade --all
Name           Id             Version       Available      Source
-----------------------------------------------------------------
Microsoft Edge Microsoft.Edge 122.0.2365.92 122.0.2365.106 winget
1 upgrades available.
> winget upgrade  Microsoft.Edge
A newer version was found, but the install technology is different from the current version installed. Please uninstall the package and install the newer version.
````

**Comment by jmcscript** [comment]
> I still have this problem... Any help?

I was able to resolve this issue, at least temporarily, by following these steps (Windows 11):

1. Open `Settings > Apps > Installed apps`
2. Click the dots menu next to **Microsoft Edge**, and then click **Modify**
3. Click **Yes** if User Account Control prompts you to modify the system
4. Click **Repair** in the Repair Microsoft Edge dialog box

**Comment by hueldoeu** [comment]
winget upgrade Microsoft.Edge never worked for me. have to open edge => help and feedback => infos about ms edge and let it update itself.

**Comment by hueldoeu** [comment]
now a new problem occured:

update problem Microsoft.Edge 130.0.2849.56 to 130.0.2849.68

Es wurde eine neuere Version gefunden, die Installationstechnologie unterscheidet sich jedoch von der aktuellen installierten Version. Deinstallieren Sie das Paket, und installieren Sie die neuere Version.

**Comment by igorskyflyer** [comment]
> now a new problem occured:
> 
> update problem Microsoft.Edge 130.0.2849.56 to 130.0.2849.68
> 
> Es wurde eine neuere Version gefunden, die Installationstechnologie unterscheidet sich jedoch von der aktuellen installierten Version. Deinstallieren Sie das Paket, und installieren Sie die neuere Version.

Translation of your comment

`German` -> `English`:
"A newer version was found, but the installation technology is different from the current installed version. Uninstall the package and install the newer version."

Also, this was an issue then as well, as mentioned by @Witchilich.

**Comment by SpecterShell** [comment]
Similar to Google Chrome (`Google.Chrome.EXE`) and Google Chrome Enterprise (`Google.Chrome`), the `Microsoft.Edge` package uses the msi installer from [Edge for Business](https://www.microsoft.com/en-us/edge/business/download), while the built-in one uses the EXE installer based on the modified version of [Chrome Setup](https://source.chromium.org/chromium/chromium/src/+/main:chrome/installer/). 

Unfortunately the latter one would not be available on WinGet because there is no static links for the EXE installers (the link comes with key and hash).

**Comment by agowa** [comment]
Why exactly cant the MSI installer just replace the EXE installer then? It clearly detects it as installed and the choice of the installer shouldn't matter then (well at least I'd expect microsoft to handle it for their chromium fork if google doesn't care at least).
